lib/fog/core/cache.rb

```diff
@@ -110,9 +110,13 @@ def self.load(model_klass, service)
       raise CacheNotFound if cache_files.empty?
 
       loaded = cache_files.map do |path|
-        model_klass = Object.const_get(load_cache(path)[:model_klass])
-        model_klass.new(load_cache(path)[:attrs])
-      end
+        if load_cache(path) # empty files, bad cache data.
+          model_klass = Object.const_get(load_cache(path)[:model_klass])
+          model_klass.new(load_cache(path)[:attrs])
+        else
+          Fog::Logger.warning("Found corrupt items in the cache: #{path}. Expire all & refresh cache soon.\n\nData:#{File.read(path)}")
+        end
```

Comment:
I wonder where this check should be done here or in the `load_cache` method instead.
This one is big enough already.